mycore-ocfl/src/main/java/org/mycore/ocfl/MCROCFLXMLMetadataManager.java

```diff
@@ -77,26 +78,38 @@ public class MCROCFLXMLMetadataManager implements MCRXMLMetadataManagerAdapter {
         Map.entry(MESSAGE_UPDATED, MCROCFLMetadataVersion.UPDATED),
         Map.entry(MESSAGE_DELETED, MCROCFLMetadataVersion.DELETED)));
 
-    private String repositoryKey = "Default";
+    private OcflRepository repository;
+
+    @MCRProperty(name = "OCFL.Repository", required = true)
+    public String repositoryKey;
+
+    /**
+     * Initializes the MetadataManager
+     */
+    public MCROCFLXMLMetadataManager() {
 
-    private static char convertMessageToType(String message) throws MCRPersistenceException {
-        if (!MESSAGE_TYPE_MAPPING.containsKey(message)) {
-            throw new MCRPersistenceException("Cannot identify version type from message '" + message + "'");
-        }
-        return MESSAGE_TYPE_MAPPING.get(message);
     }
 
-    public OcflRepository getRepository() {
-        return MCROCFLRepositoryProvider.getRepository(getRepositoryKey());
+    public MCROCFLXMLMetadataManager(String repositoryKey) {
+        this.repositoryKey = repositoryKey;
+        repository = MCROCFLRepositoryProvider.getRepository(repositoryKey);
     }
 
-    public String getRepositoryKey() {
-        return repositoryKey;
+    @MCRPostConstruction
+    @SuppressWarnings("PMD")
+    public void init(String property) {
+        repository = MCROCFLRepositoryProvider.getRepository(repositoryKey);
     }
```

Comment:
See comment on the MCROCFLXMLUserManager regarding performance impact of initializing it with `new MCROCFL[...]Manager()`.